website/content/docs/v0.9/Guides/disk-encryption.md

```diff
@@ -0,0 +1,177 @@
+---
+title: "Disk Encryption"
+description: "This page contains guides related kernel level disk encryption"
+---
+
+LUKS2 disk encryption is supported by the Linux kernel modules and `cryptsetup` utility.
+
+Sequencer runs additional steps when encryption is enabled.
+If the disk encryption is enabled for the STATE partition, the sequencer will:
+
+- save STATE encryption config as JSON in the META partition.
+- before mounting the STATE partition, load encryption configs either from the machine config or from the META partition,
+  the machine config is always preferred over the META one.
+- before mounting the STATE partition, format and encrypt it, but only if it is empty and has no filesystem.
+
+If the disk encryption is enabled for the EPHEMERAL partition, the sequencer will:
+
+- get the encryption config from the machine config.
+- before mounting the EPHEMERAL partition, encrypt and format it, but only if it is empty and has not filesystem.
+
+## Configuration
```

Comment:
we should talk about encryption key providers?

Reply:
We have it there